website/addons/base/__init__.py

```diff
@@ -592,11 +592,13 @@ def verify_oauth_access(self, node, external_account, metadata=None):
 
     def get_nodes_with_oauth_grants(self, external_account):
         # Generator of nodes which have grants for this external account
-        return (
-            Node.load(node_id)
-            for node_id, grants in self.oauth_grants.iteritems()
-            if external_account._id in grants.keys()
-        )
+        nodes = []
+        for node_id, grants in self.oauth_grants.iteritems():
```

Comment:
Why was this changed to a loop. Why not just add the check for `not node.is_deleted` to the generator expression?